packages/tc-markdown-parser/__tests__/property-value.spec.js

```diff
@@ -0,0 +1,302 @@
+		# name
+
+		## some enum
+		first
+
+	`);
+
+	expect(data.someEnum).toBe('First');
+});
+
+test('enums types with incorrect values produce error', async () => {
+	const { data, errors } = await parser.parseRunbookString(here`
+		# name
+
+		## some enum
+
+		Fourth
+	`);
+
+	expect(data.someEnum).toBe(undefined);
+	expect(errors.length).toBe(1);
+	const [{ message }] = errors;
+	expect(message).toMatch(/\bfourth\b/);
+	expect(message).toContain('AnEnum');
+});
+
+// Before this test, we had links coming out wrapped in triangle brackets
+test('urls should stay urls', async () => {
+	const { data } = await parser.parseRunbookString(here`
+		# name
+
+		## some url
+		https://snoot.club
+	`);
+
+	expect(data.someUrl).toBe('https://snoot.club');
+	expect(typeof data.someUrl).toBe('string');
+});
+
+test('nested fields are coerced to string (the code)', async () => {
+	const { data } = await parser.parseRunbookString(here`
+		# name
+
+		## favourite child
+
+		cheerabbits
+	`);
+
+	expect(data.favouriteChild).toBe('cheerabbits');
+});
+
+test('properties with hasMany turn bulleted lists into arrays', async () => {
+	const { data, errors } = await parser.parseRunbookString(here`
+		# name
+
+		## children
+
+		* chee-rabbits
+
+		## younger siblings
+
+		* ft-app-fruitcake
+		* apple-quicktime
+	`);
+
+	expect(errors).toHaveLength(0);
+	expect(data.children).toEqual(['chee-rabbits']);
+	expect(data.youngerSiblings).toEqual(['ft-app-fruitcake', 'apple-quicktime']);
+});
+
+test.skip('dependents is banned', async () => {
+	const { data, errors } = await parser.parseRunbookString(here`
+		# name
+
+		## children
+
+		* chee-rabbits
+
+		## dependents
+
+		* ft-app-fruitcake
+	`);
+
+	expect(errors).toHaveLength(1);
+	expect(data.children).toEqual(['chee-rabbits']);
+	const [{ message }] = errors;
+	expect(message).toEqual(
+		'dependents is not permitted within runbook.md (to allow other people to edit it)',
+	);
+});
+
+test('properties with hasMany must be bulleted lists', async () => {
+	const { errors } = await parser.parseRunbookString(here`
+		# name
+
+		## children
+
+		chee-rabbits
+	`);
+
+	expect(errors).toHaveLength(1);
+
+	const [{ message }] = errors;
+	expect(message).toContain('list');
+	expect(message).toContain('bullet');
+});
+
+test('properties without hasMany must not be bulleted lists', async () => {
+	const { errors } = await parser.parseRunbookString(here`
+		# name
+
+		## some url
+
+		* https://ft.com
+	`);
+
+	expect(errors).toHaveLength(1);
+
+	const [{ message }] = errors;
+	expect(message).toContain('list');
+	expect(message).toContain('bullet');
+});
+
+test('subdocuments have their headers reduced two levels', async () => {
+	const { data } = await parser.parseRunbookString(here`
+		# name
+
+		## some document
+
+		### hello
+	`);
+
+	expect(data).toEqual({
+		name: 'name',
+		someDocument: '# hello',
+	});
+});
+
+test.skip('last review date is banned', async () => {
```

Comment:
because `lastReviewDate` is not defined in MainType

Reply:
can delete this test